extensions/ql-vscode/src/data-extensions-editor/queries/java.ts

```diff
@@ -29,6 +29,100 @@ where
   supported = isSupported(api) and
   usage = aUsage(api)
 select usage, apiName, supported.toString(), "supported", api.jarContainer(), "library"
+`,
+  frameworkModeQuery: `/**
+ * @name Public methods
+ * @description A list of APIs callable by consumers. Excludes test and generated code.
+ * @tags telemetry
+ * @kind problem
+ * @id java/telemetry/fetch-public-methods
+ */
+
+import java
+private import semmle.code.java.dataflow.DataFlow
+private import semmle.code.java.dataflow.ExternalFlow
+private import semmle.code.java.dataflow.FlowSources
+private import semmle.code.java.dataflow.FlowSummary
+private import semmle.code.java.dataflow.internal.DataFlowPrivate
+private import semmle.code.java.dataflow.internal.FlowSummaryImpl as FlowSummaryImpl
+private import semmle.code.java.dataflow.internal.ModelExclusions
+
+/** Holds if the given callable is not worth supporting. */
+private predicate isUninteresting(Callable c) {
+  c.getDeclaringType() instanceof TestLibrary or
+  c.(Constructor).isParameterless()
+}
+
+class PublicMethod extends Callable {
```

Comment:
For my understanding: this is basically a copy of `ExternalAPI` with the constructor changed? I cannot completely parse whether that is exactly what we want, but I think at a minimum it is close.

I think moving to the CodeML queries is something we should do, otherwise we probably want to make some changes (e.g. sharing the code for all those methods).

Reply:
Yes, this basically changes `not this.fromSource()` to `this.isPublic()` in the constructor. The rest of the methods are very similar. We can work on reducing the duplication in a follow-up.